haystack/nodes/search_engine/web.py

```diff
@@ -0,0 +1,133 @@
+import pydoc
+from typing import List, Dict, Any, Optional, Union, Tuple, Type, Callable
+
+from haystack import BaseComponent, MultiLabel, Document
+from haystack.nodes import PromptNode, PromptTemplate, TopPSampler
+from haystack.nodes.search_engine.base import SearchEngine
+
+
+class WebSearch(BaseComponent):
+    """
+    WebSearch queries a search engine and retrieve results. It can use various search engine providers, e.g. SerperDev,
+    SerpAPI, etc.
+    """
+
+    outgoing_edges = 1
+
+    def __init__(self, api_key: str, search_engine_provider: Union[str, SearchEngine] = "SerperDev", **kwargs):
+        """
+        :param api_key: API key for the search engine provider
+        :param search_engine_provider: Name of the search engine provider, e.g. "SerperDev", "SerpAPI"
+        :param kwargs: Additional parameters to pass to the search engine provider
```

Comment:
what are the parameters?

Reply:
They could be anything, as each search provider has their custom parameters.